sdk/CorrelationContext.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Sdk;
+
+use OpenTelemetry\Context\Context;
+use OpenTelemetry\Context\ContextKey;
+
+class CorrelationContext
+{
+    use Context;
+
+    /**
+     * @param CorrelationContext $context
+     */
+    public function getCorrelations($context = null)
+    {
+        // TODO: Write me
+    }
+
+    /**
+     * @param ContextKey $key
+     *
+     * @return CorrelationContext
+     */
+    public function removeCorrelation(ContextKey $key): CorrelationContext
+    {
+        if ($this->key === $key) {
+            return $this->parent;
+        }
+
+        $this->removeCorrelationHelper($key, null);
+
+        return $this;
+    }
+
+    /**
+     * @param ContextKey $key
+     * @param CorrelationContext|null $child
+     * @suppress PhanUndeclaredMethod
+     * @suppress PhanTypeMismatchArgument
```

Comment:
I added this to appease the CI for now, but I think we might need to address it.

Reply:
One possible solution to this and the `PhanUndeclaredMethod` is to add:

```
    /**
     * @var CorrelationContext|null
     */
    private $parent;
```
to the top of `CorrelationContext` which I think kinda makes sense... within a `CorrelationContext`, we know because of the structure of the logic that `$this->parent` will always be a `CorrelationContext` but the type system cannot assert this. 

I'm looking at alternatives, specifically if adding an inline type check would be better

Another option is to change this function to look like this:
```
    private function removeCorrelationHelper(ContextKey $key, ?CorrelationContext $child)
    {
        $parent = $this->parent;
        '@phan-var CorrelationContext|null $parent';
        if ($this->key !== $key) {
            if (null === $parent) {
                return;
            }

            $this->parent->removeCorrelationHelper($key, $this);
            return;
        }

        $child->setParent($parent);
    }

```

But I kinda hate that... its very weird.


The other option is to add a type assertion to the helper:

```
    private function removeCorrelationHelper(ContextKey $key, ?CorrelationContext $child)
    {
+       assert(($this->parent instanceof CorrelationContext) || (is_null($this->parent)));
        if ($this->key !== $key) {
            if (null === $this->parent) {
                return;
            }
        ...
```

I might hate this one the least? I dunno, that or just mirroring the `$parent` into the top of `CorrelationContext` I dunno